### PR TITLE
Issue #27 setup data source

### LINKED ISSUE #27: Setup DataSource
You need to setup DataSource, so our project can connect to PostgreSQL and execute queries.

You can use JDBC driver for that.

### PR DESCRIPTION
**Pull request by nikaalin**

### CODE REVIEW

**Review by JustRevolt — APPROVED** [review]

**Review by ChirkinaTanya — APPROVED** [review]

**Review by a-deriabin — APPROVED** [review]

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorInterface.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.ifmo.epampractice.DAO;
+
+import javax.sql.DataSource;
+import java.io.IOException;
+
+public interface DBConnectorInterface {
+    void setProperties(String url) throws IOException;
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/DAO/DBConnectorInterface.java]: Are you sure you want to setup path to property file for each instance?

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.ifmo.epampractice.DAO;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.postgresql.ds.PGSimpleDataSource;
+
+public class DBConnectorPostgres implements DBConnectorInterface {
+    Properties properties = new Properties();
+    PGSimpleDataSource dataSource = null;
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java]: `= null ` not necessary and can be private

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.ifmo.epampractice.DAO;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.postgresql.ds.PGSimpleDataSource;
+
+public class DBConnectorPostgres implements DBConnectorInterface {
+    Properties properties = new Properties();
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java]: can be private

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    PGSimpleDataSource dataSource = null;
+
+    public DBConnectorPostgres() {
+        try {
+            dataSource = new PGSimpleDataSource();
+            setProperties("/properties/dbPostgres.properties");
+        } catch (Exception e) {
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java]: you can specify error IOException

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void setProperties(String url) throws IOException {
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java]: exclude from interface and make it private

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package com.ifmo.epampractice.DAO;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.postgresql.ds.PGSimpleDataSource;
+
+public class DBConnectorPostgres implements DBConnectorInterface {
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java]: make it singleton

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    Properties properties = new Properties();
+    PGSimpleDataSource dataSource = null;
+
+    public DBConnectorPostgres() {
+        try {
+            dataSource = new PGSimpleDataSource();
+            setProperties("/properties/dbPostgres.properties");
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java]: `/properties/dbPostgres.properties` make it constant

File: src/main/resources/properties/dbPostgres.properties [hunk 8]
```diff
@@ -0,0 +1,4 @@
+DRIVER=org.postgresql.Driver
+URL=jdbc:postgresql://localhost:5432/postgres
+USERNAME=postgres
+PASSWORD=rootroot
```
Comment (PavelEzhkov) [inline src/main/resources/properties/dbPostgres.properties]: add new line

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    public void setProperties(String url) throws IOException {
+        properties.load(this.getClass().getResourceAsStream(url));
+        dataSource.setUrl(properties.getProperty("URL"));
+        dataSource.setUser(properties.getProperty("USERNAME"));
+        dataSource.setPassword(properties.getProperty("PASSWORD"));
+
+    }
+
+    @Override
+    public PGSimpleDataSource getDataSource() {
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java]: let's return Connection

**Review by PavelEzhkov — COMMENTED** [review]
rename folder dao instead DAO

File: src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java (hunk 9 above)
Comment (nikaalin) [reply src/main/java/com/ifmo/epampractice/DAO/DBConnectorPostgres.java]: done

**Review by Nursatb — APPROVED** [review]

**Review by PavelEzhkov — APPROVED** [review]

**Pull request merged**